spot_ros2_control/hardware/src/spot_hardware_interface.cpp

```diff
@@ -0,0 +1,356 @@
+// File modified. Modifications Copyright (c) 2024 Boston Dynamics AI Institute LLC.
+// All rights reserved.
+
+// --------------------------------------------------------------
+// Copyright 2020 ros2_control Development Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "spot_ros2_control/spot_hardware_interface.hpp"
+
+#include <chrono>
+#include <cmath>
+#include <limits>
+#include <memory>
+#include <vector>
+
+#include "hardware_interface/types/hardware_interface_type_values.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+namespace spot_ros2_control {
+
+void StateStreamingHandler::handle_state_streaming(::bosdyn::api::RobotStateStreamResponse& robot_state) {
+  const auto& position_msg = robot_state.joint_states().position();
+  const auto& velocity_msg = robot_state.joint_states().velocity();
+  const auto& load_msg = robot_state.joint_states().load();
+
+  current_position_.assign(position_msg.begin(), position_msg.end());
+  current_velocity_.assign(velocity_msg.begin(), velocity_msg.end());
+  current_load_.assign(load_msg.begin(), load_msg.end());
+}
+
+const std::vector<float>& StateStreamingHandler::get_position() const {
+  return current_position_;
+}
+const std::vector<float>& StateStreamingHandler::get_velocity() const {
+  return current_velocity_;
+}
+const std::vector<float>& StateStreamingHandler::get_load() const {
+  return current_load_;
+}
```

Comment:
@khughes-bdai meta: splitting accessors like this will make it harder to synchronize read and writes. Returning a complete `struct` by value will simplify it.